tidy3d/components/heat/simulation.py

```diff
@@ -0,0 +1,815 @@
+
+        return plot_params
+
+    def _plot_boundary_condition(
+        self, shape: Shapely, boundary_spec: HeatBoundarySpec, ax: Ax
+    ) -> Ax:
+        """Plot a structure's cross section shape for a given boundary condition."""
+        plot_params_bc = self._get_bc_plot_params(boundary_spec=boundary_spec)
+        ax = self.plot_shape(shape=shape, plot_params=plot_params_bc, ax=ax)
+        return ax
+
+    @staticmethod
+    def _structure_to_bc_spec_map(
+        plane: Box, structures: Tuple[Structure, ...], boundary_spec: Tuple[HeatBoundarySpec, ...]
+    ) -> Dict[str, HeatBoundarySpec]:
+        """Construct structure name to bc spec inverse mapping. One structure may correspond to
+        multiple boundary conditions."""
+
+        named_structures_present = {structure.name for structure in structures if structure.name}
+
+        struct_to_bc_spec = {}
+        for bc_spec in boundary_spec:
+            bc_place = bc_spec.placement
+            if (
+                isinstance(bc_place, (StructureBoundary, StructureSimulationBoundary))
+                and bc_place.structure in named_structures_present
+            ):
+                if bc_place.structure in struct_to_bc_spec:
+                    struct_to_bc_spec[bc_place.structure] += [bc_spec]
+                else:
+                    struct_to_bc_spec[bc_place.structure] = [bc_spec]
+
+            if isinstance(bc_place, StructureStructureInterface):
+                for structure in bc_place.structures:
+                    if structure in named_structures_present:
+                        if structure in struct_to_bc_spec:
+                            struct_to_bc_spec[structure] += [bc_spec]
+                        else:
+                            struct_to_bc_spec[structure] = [bc_spec]
+
+            if isinstance(bc_place, SimulationBoundary):
+                struct_to_bc_spec[HEAT_BACK_STRUCTURE_STR] = [bc_spec]
+
+        return struct_to_bc_spec
+
+    @staticmethod
+    def _medium_to_bc_spec_map(
+        plane: Box, structures: Tuple[Structure, ...], boundary_spec: Tuple[HeatBoundarySpec, ...]
+    ) -> Dict[str, HeatBoundarySpec]:
+        """Construct medium name to bc spec inverse mapping. One medium may correspond to
+        multiple boundary conditions."""
+
+        named_mediums_present = {
+            structure.medium.name for structure in structures if structure.medium.name
+        }
+
+        med_to_bc_spec = {}
+        for bc_spec in boundary_spec:
+            bc_place = bc_spec.placement
+            if isinstance(bc_place, MediumMediumInterface):
+                for med in bc_place.mediums:
+                    if med in named_mediums_present:
+                        if med in med_to_bc_spec:
+                            med_to_bc_spec[med] += [bc_spec]
+                        else:
+                            med_to_bc_spec[med] = [bc_spec]
+
+        return med_to_bc_spec
+
+    @staticmethod
```

Comment:
it will eventually go to `components/unstructured_sim`

Reply:
a lot of plotting logic probably can be separated into other files, since it seems like a lot of them are just `staticmethod`s. But we can do this at a later time, I think.